indico/modules/events/registration/client/js/form/fields/BooleanInput.jsx

```diff
@@ -6,41 +6,79 @@
 // LICENSE file for more details.
 
 import PropTypes from 'prop-types';
-import React from 'react';
+import React, {useState} from 'react';
 import {useSelector} from 'react-redux';
+import {Button, Form, Label} from 'semantic-ui-react';
 
-import {FinalDropdown} from 'indico/react/forms';
+import {FinalDropdown, FinalInput, validators as v} from 'indico/react/forms';
 import {Translate} from 'indico/react/i18n';
 
 import {getCurrency} from '../../form_setup/selectors';
 
-export default function BooleanInput({htmlName, disabled, isRequired, defaultValue, price}) {
+import {PlacesLeft} from './PlacesLeftLabel';
+
+import '../../../styles/regform.module.scss';
+
+export default function BooleanInput({
+  disabled,
+  title,
+  isRequired,
+  defaultValue,
+  price,
+  placesLimit,
+}) {
+  const [value, setValue] = useState(defaultValue);
   const currency = useSelector(getCurrency);
+  const makeOnClick = newValue => () => {
+    if (value === newValue && !isRequired) {
+      setValue('');
+    } else {
+      setValue(newValue);
+    }
+  };
 
   return (
-    <select name={htmlName} disabled={disabled} defaultValue={defaultValue}>
-      {!isRequired && <option value="">{Translate.string('Choose an option')}</option>}
-      <option value="yes">
-        {Translate.string('Yes')} {!!price && `(${price} ${currency})`}
-      </option>
-      <option value="no">{Translate.string('No')}</option>
-    </select>
+    <Form.Field required={isRequired} disabled={disabled} styleName="field">
+      <label>{title}</label>
```

Comment:
We're repeating this all the time now. Can't we move it to the `FormItem` (or another component) to cover the common cases? If some of the fields require special logic we can always make it opt-out via the field type meta...